spirv_msl.cpp

```diff
@@ -5643,6 +6207,228 @@ void CompilerMSL::emit_instruction(const Instruction &instruction)
 	previous_instruction_opcode = opcode;
 }
 
+// If the underlying resource has been used for comparison then duplicate loads of that resource must be too
+static inline bool image_opcode_is_sample_no_dref(Op op)
+{
+	switch (op)
+	{
+	case OpImageSampleExplicitLod:
+	case OpImageSampleImplicitLod:
+	case OpImageSampleProjExplicitLod:
+	case OpImageSampleProjImplicitLod:
+	case OpImageFetch:
+	case OpImageRead:
+	case OpImageSparseSampleExplicitLod:
+	case OpImageSparseSampleImplicitLod:
+	case OpImageSparseSampleProjExplicitLod:
+	case OpImageSparseSampleProjImplicitLod:
+	case OpImageSparseFetch:
+	case OpImageSparseRead:
+		return true;
+		
+	default:
+		return false;
+	}
+}
+
+void CompilerMSL::emit_texture_op(const Instruction &i)
+{
```

Comment:
Was this rebased on the code before YCbCr support was added a few weeks ago? This code should not exist anymore.

Reply:
I simplified this overriden function to also make use of `to_texture_op`, but still preserve the use of the new option `ios_use_framebuffer_fetch_subpasses`.

Reply:
`shaders-msl/asm/frag/frame-buffer-fetch-subpass-input.asm.frag` is not tagged with `.ios.` so it is compiled as macOS, and the test is a bit moot. See `cross_compile_msl` in `test_shaders.py` how arguments are passed to the CLI tool when running tests.

I think this function can be simplified quite a bit. There is no need to inherit expression dependencies (it's just an input value) and control dependent expressions (no LOD). Probably best to do something like `if (frame buffer fetch case) { do_thing; } else { CompilerGLSL::emit_texture_op(i); }`